### PR TITLE
better display of errors on workspace and submission list pages [risk: low]

### PR DESCRIPTION
**Pull request by davidangb**
If the user encountered an error loading workspaces:
* the error message was likely hidden from view
* when shown, the visual display of the error message was cramped.

If the user encountered an error loading submissions in a workspace:
* the error message was hidden from view

# Workspace List
_most errors were hidden:_
wserror-hidden

_when shown, error message was cramped:_
wserror-notpadded

_this PR fixes those (visual display is still minimalist):_
wserror-padded

# Submission List
_errors were hidden:_
sub-list-error-hidden

_this PR fixes it:_
subb-list-error-detail
 (visual display is still minimalist):_

-----

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/page/workspaces_list.cljs [hunk 1]
```diff
@@ -428,7 +428,11 @@
           [WorkspaceTable
            (merge props (utils/restructure workspaces featured-workspaces))])
         {:loading-text "Loading workspaces..."
-         :rephrase-error #(get-in % [:parsed-response :workspaces :error-message])})))
+         :rephrase-error (fn [ajax-response]
+                           (let [msg (or
+                                      (get-in ajax-response [:parsed-response :message])
+                                      (get-in ajax-response [:parsed-response :workspaces :error-message]))]
+                            [:div {:style {:padding "50px"}} msg]))})))
```
Comment (davidangb) [inline src/cljs/main/broadfcui/page/workspaces_list.cljs:435]: from reverse-engineering the backend code, I don't see anything that delivers an error message in `[:parsed-response :workspaces :error-message]`. But, it felt safe enough to leave that path in, as the second clause in the `(or)` statement.

File: src/cljs/main/broadfcui/utils/ajax.cljs [hunk 2]
```diff
@@ -139,3 +139,15 @@
...
+  ([get-parsed-response status-text status-code] (extract-error-message get-parsed-response status-text status-code []))
+  ([get-parsed-response status-text status-code key-path]
+    (let [default-error (str status-code " " status-text)
+          response (try
+                     (get-parsed-response)
+                     (catch js/Exception e {:message default-error}))]
+      (or
+        (if (not-empty key-path) (get-in response key-path))
+        (:message response)
+        (default-error)))))
```
Comment (davidangb) [inline src/cljs/main/broadfcui/utils/ajax.cljs]: I probably didn't need to make this a generic error-extraction function; I could have done it all in the handler for the submission list. I'm hoping to use this elsewhere in the future.

**Comment by davidangb** [comment]
test run was hung in jenkins. let's try again: jenkins retest

File: src/cljs/main/broadfcui/utils/ajax.cljs [hunk 3]
```diff
@@ -139,3 +139,15 @@
...
+
+(defn extract-error-message
+  ([get-parsed-response status-text status-code] (extract-error-message get-parsed-response status-text status-code []))
+  ([get-parsed-response status-text status-code key-path]
+    (let [default-error (str status-code " " status-text)
+          response (try
+                     (get-parsed-response)
+                     (catch js/Exception e {:message default-error}))]
+      (or
+        (if (not-empty key-path) (get-in response key-path))
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/utils/ajax.cljs]:
```clojure
(when (seq key-path) (get-in response key-path))
```

File: src/cljs/main/broadfcui/utils/ajax.cljs (hunk 2 above)
Comment (MattPutnam) [inline src/cljs/main/broadfcui/utils/ajax.cljs]: No parens

**Comment by davidangb** [comment]
unit tests failed in jenkins with
```
Could not find artifact binaryage:env-config:jar:0.2.2 in central (https://repo1.maven.org/maven2/)
Could not transfer artifact binaryage:env-config:jar:0.2.2 from/to clojars (https://repo.clojars.org/): Failed to transfer file: https://repo.clojars.org/binaryage/env-config/0.2.2/env-config-0.2.2.jar. Return code is: 503 , ReasonPhrase:Connection timed out.
This could be due to a typo in :dependencies, file system permissions, or network issues.
```
I assume this is transient.

**Comment by davidangb** [comment]
automation tests passed but we had that transient failure in unit tests. For the sake of completeness: jenkins retest.

**Review by MattPutnam — APPROVED** [review]
👍

**Pull request merged**